### ISSUE TITLE
emscripten 1.37.37 broken on OSX

### ISSUE THREAD

**Issue by kumpera** [kripken/emscripten#6456]
The C compiler tries to run python2, but that's not available on OSX systems. The right name of the binary is python2.7 (but probably requires some detection)

**Comment by kripken** [comment]
Thanks for filing, do you have a stack trace of where this happens? I wonder if it's a specific place with a problem. Or, is this that in the `.emscripten` file that is generated on first run emscripten gets it wrong and uses `python2` instead of `python2.7`?

**Comment by kripken** [comment]
Sounds like this is a recent regression - would be good to confirm it's between 1.37.36 and 1.37.37, and if possible to bisect into that range.

**Comment by brion** [comment]
With 1.37.37 installed via emsdk, I see some environment bits set via `emconfigure` such as:

```
CC=/Users/brion/src/emsdk-portable/emscripten/1.37.37/emcc.py
LD=/Users/brion/src/emsdk-portable/emscripten/1.37.37/emcc.py
LDSHARED=/Users/brion/src/emsdk-portable/emscripten/1.37.37/emcc.py
```

If I look at the `emcc.py`, I notice it's explicitly using python2:

```
#!/usr/bin/env python2
```

which thus fails whenever the $CC or $LD are invoked.

I seem to remember there being a change to the paths for invoking things in `emconfigure` recently, which may be related.

**Comment by brion** [comment]
1.37.36, also installed via emsdk, has

```
CC=python /Users/brion/src/emsdk-portable/emscripten/1.37.36/emcc.py
```

which works on macOS.

**Comment by kripken** [comment]
Thanks! Looking in that range, the best guess commits seem to be 9a5a9b6224b79eafb33517226bdb42c6acbd137f

I don't have a Mac to test on - would be good to confirm if that's the bad commit. It's not obvious to me what's wrong there, especially since that part of the commit reverts us back to a previous state which I presume was good (we stayed on it for a long time). Odd.

**Comment by kumpera** [comment]
I can confirm 1.37.36 works on OSX just fine. I switched mono to use it yesterday to workaround this issue.

**Comment by brion** [comment]
Yeah I can see what's going wrong there, at https://github.com/kripken/emscripten/commit/9a5a9b6224b79eafb33517226bdb42c6acbd137f#r28630452

**Comment by kripken** [comment]
Proposed fix in #6460.

**Comment by floooh** [comment]
Btw, I worked around this with ```brew install python@2```, this installs a ```python2``` link in the path with a python2.7.14 installation behind it.
